lib/logstash/devutils/rspec/logstash_helpers.rb

```diff
@@ -36,32 +37,45 @@ def sample(sample_event, &block)
         pipeline.instance_eval { @filters.each(&:register) }
 
         event.each do |e|
-          pipeline.filter(e) {|new_event| results << new_event }
+          pipeline.filter(e) {|new_event| results << new_event}
         end
 
         pipeline.flush_filters(:final => true) do |e|
           results << e unless e.cancelled?
         end
 
-        results
+        results.select{|e| !e.cancelled?}
       end
 
-      subject { results.length > 1 ? results: results.first }
+      subject { results.length > 1 ? results : results.first }
```

Comment:
In order to be uniform, is good to remove the space after the brackets in here, isn't?